_posts/2021-08-13-multi-factor-authentication-with-twilio-verify-api.md

```diff
@@ -0,0 +1,88 @@
+slug: "multi-factor-authentication-dev-perspective-on-working-with-twilio-verify-api"
+category: blog
+post_image: /assets/images/multi-factor-authentication/header_image.png
+author: seosamh
+date: 2021-08-13 09:30:00
+excerpt: "Multi Factor Authentication is a method of authentication that adds extra steps of verification for gaining access to resources such as an online account, internal system, application or network. Authentication has been traditionally completed by validating a username and password."
+graphic: /assets/images/multi-factor-authentication/header_image.png
+---
+
+### What is Multi Factor Authentication (MFA)?
+
+Multi Factor Authentication is a method of authentication that adds extra steps of verification for gaining access to resources such as an online account, internal system, application or network. Authentication has been traditionally completed by validating a username and password. However, there is now a need for higher security as cybercrime increases and evolves. Techniques such as brute force attacks, and social engineering have weakened conventional security measures for restriction of access, and they no longer provide sufficient guard against vulnerability. There are also regulations such as PSD2 in the European Union that place an onus on certain industries to ensure enforcement of strong customer authentication.
+
+As a Software Engineer at OnePageCRM, I have recently worked on the implementation of Two Step Authentication (2SA), which is a type of Multi Factor Authentication. The first step in completing this project was to consider which factors provide the greatest security for our users during the login process.
+
+The options considered included:
+
+**SMS Token Authentication / Phone Authentication:** The use of a third-party API which sends users an SMS message or automated phone call during the authentication process. The user is then required to enter a pin or carry out an alternative action such as pressing a key.  API options include GatewayAPI, Twilio API, Messagebird API, etc.
+
+**Hardware Token Authentication:** This involves using a separate purpose-built device to generate “One Time Passwords”. This is considered one of the most secure ways of providing an extra step but may prove impractical and unnecessary in most settings.
+
+**Software Token Authentication:** Authentication applications such as Google Authenticator and Authy on mobile devices provide similar security to a hardware token whereby a “One Time Password” is provided.
+
+**Biometric Verification:** Provide authentication through the users’ face or fingerprint. However, it’s important to note that not all users have access to devices capable of providing this functionality.
+
+**Email Token Verification:** Use email address to verify attempts to gain access.
+
+Let’s examine a sample use case of the Twilio Verify API to provide an extra factor with a user’s phone number at a high level
```

Comment:
Extra **step**. 

Username and password is a knowledge factor
A 6 digit code sent via SMS is also a knowledge factor. One reason for this is because in order to change the phone number associated with a sim card may only require a small amount of "knowledge" and not actual possesion.